viewer/packages/sector-parser/src/GltfSectorParser.ts

```diff
@@ -2,224 +2,289 @@
-    this._gltfReader.registerExtensions([MeshGPUInstancing]);
+    this._glbMetadataParser = new GlbMetadataParser();
   }
 
-  public parseSector(data: ArrayBuffer): TypedGeometry[] {
-    const document: Document = this._gltfReader.readBinary(data);
+  public parseSector(data: ArrayBuffer) {
+    const headers = this._glbMetadataParser.parseGlbMetadata(data);
 
-    const defaultScene = document.getRoot().getDefaultScene();
-    const typedGeometryBuffers: TypedGeometry[] = [];
+    const json = headers.json;
 
-    defaultScene!.traverse((node: Node) => {
-      const processedNode = this.processNode(node);
+    return this.traverseDefaultSceneNodes(json, headers, data);
+  }
 
-      if (processedNode) typedGeometryBuffers.push(processedNode);
-    });
+  private traverseDefaultSceneNodes(json: GltfJson, headers: GlbHeaderData, data: ArrayBuffer) {
+    const typedGeometryBuffers: { type: RevealGeometryCollectionType; buffer: THREE.BufferGeometry }[] = [];
+
+    const defaultSceneNodeIds = json.scenes[json.scene].nodes;
+
+    defaultSceneNodeIds
+      .map(nodeId => json.nodes[nodeId])
+      .forEach(node => {
+        const processedNode = this.processNode(node, headers, data)!;
+
+        if (processedNode === undefined) {
+          return;
+        }
+
+        typedGeometryBuffers.push(processedNode);
+      });
 
     return typedGeometryBuffers;
   }
 
-  private processNode(node: Node): TypedGeometry | null {
-    const instanced = node.getExtension<InstancedMesh>(MeshGPUInstancing.EXTENSION_NAME);
-    const mesh = node.getMesh();
-    let instanceId: number | undefined;
+  private processNode(node: Node, glbHeaderData: GlbHeaderData, data: ArrayBuffer) {
+    const instancingExtension = node.extensions?.EXT_mesh_gpu_instancing;
+    const meshId = node.mesh;
+
+    if (instancingExtension === undefined && meshId === undefined) return undefined;
 
-    if (this.isEmptyNode(instanced, mesh)) return null;
+    const bufferGeometry = instancingExtension ? new THREE.InstancedBufferGeometry() : new THREE.BufferGeometry();
 
-    const bufferGeometry = instanced ? new THREE.InstancedBufferGeometry() : new THREE.BufferGeometry();
+    const geometryType = RevealGeometryCollectionType[node.name as keyof typeof RevealGeometryCollectionType];
 
-    // Casts the string (node.getName()) to a RevealGeometryCollectionType enum (e.g. BoxCollection)
-    const geometryType = RevealGeometryCollectionType[node.getName() as keyof typeof RevealGeometryCollectionType];
+    const payload: GeometryProcessingPayload = {
+      bufferGeometry,
+      geometryType,
+      glbHeaderData,
+      instancingExtension: instancingExtension,
+      meshId,
+      data
+    };
 
     switch (geometryType) {
       case RevealGeometryCollectionType.InstanceMesh:
-        instanceId = this.processInstancedMesh(mesh, instanced, bufferGeometry);
+        //   instanceId = this.processInstancedMesh(mesh, instanced, bufferGeometry);
```

Comment:
I don't think it's a good idea to merge this PR before we have instance meshes and feature parity with as is (?)